cmd/feed-dns/main.go

```diff
@@ -67,6 +69,10 @@ func init() {
 		"A label=value pair to attach to metrics pushed to prometheus. Specify multiple times for multiple labels.")
 	flag.IntVar(&awsAPIRetries, "aws-api-retries", defaultAwsAPIRetries,
 		"Number of times a request to the AWS API is retried.")
+	flag.StringVar(&internalCNAME, "internal-cname", "",
+		"the CNAME of the internal facing load-balancer. If specified, external-cname must also be given.")
+	flag.StringVar(&externalCNAME, "external-cname", "",
+		"the CNAME of the internet facing load-balancer. If specified, internal-cname must also be given.")
```

Comment:
I think it'd be better if we did this as a mapping, so you could map internet-scheme to hostname in case we wanted more than the two ELB types. Or if we wanted to use different names.

Maybe that's overcomplicating things though, I'm not sure. The annotation sky.uk/frontend-scheme can be anything though, which seems to suggest this should be parameterised as a map.

Reply:
Also - wouldn't we want to support multiple IPs for redundancy?

Reply:
The annotation's value can't be anything; in the source it's very much tied to just 'internal' or 'internet-facing'.

Reply:
The scheme is limited to `internal` and `internet-facing` (which is an ELB concept like you say above) and although the annotation has been renamed to be more general I don't think there's a need to extend it any further.